### DISCUSSION TITLE
Communication ower ESP-NOW or ESP LR.

### DISCUSSION THREAD

**Discussion by CutePotatoDev** [markqvist/Reticulum#188] [General]
Thinking or it's possible to set up RNode and Reticilum to work over ESP-NOW or ESP LR.
In my mind i have as cheap as possible standalone communication device. And esp32 without any extra LoRa transceiver sounds interesting starting point.
It's possible or not to much ?
How useful or pointless be this kind of usage ?

**Comment by markqvist** [comment]
Yes, that feature is very useful and valuable indeed, it's been on the [roadmap](https://github.com/markqvist/Reticulum/blob/master/Roadmap.md#interfaceability) for a while already :)

**Comment by CutePotatoDev** [comment]
As i understand from checking code, RNode working only as data proxy. Exchange lora part to ESP-NOW in theory not to much complicated change.

But more interesting part for my is Reticilum. Or at all in future possible port it from Python to C/C++ and install to esp together with RNode. To have cheap ass and compact device.

My final idea to have something similar to meshtastic canned message module. 
Like old style mobile phone, with screen and keyboard. And as possible lower price.

**Comment by dmahony** [comment]
Another option using ESP32, Bluetooth Coded Phy Long Range mode! Do you think it is possible @markqvist ?

**Reply by markqvist** [reply]
Almost everything is *possible*, it depends on how much time and resources I have to actually research and then implement it ;)
